pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -220,6 +221,16 @@ public boolean setReplicated(boolean replicated) {
 
     @Override
     public CompletableFuture<Void> addConsumer(Consumer consumer) {
+        CompletableFuture<Void> inProgressResetCursorFuture = this.inProgressResetCursorFuture;
+        if (inProgressResetCursorFuture != null) {
+            return inProgressResetCursorFuture.thenApply(ignore -> null)
+                    .exceptionally(ex -> null).thenCompose(ignore -> addConsumerInternal(consumer));
```

Comment:
`.thenApply(ignore -> null).exceptionally(ex -> null)` could be replaced with `.handle((ignore, ex) -> null)` to make it shorter.

Reply:
Improved, thanks for your suggestion